Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp

```diff
@@ -161,12 +161,35 @@ void SVGFEConvolveMatrixElement::setKernelUnitLength(float x, float y)
     updateSVGRendererForElementChange();
 }
 
+bool SVGFEConvolveMatrixElement::isValidTargetXOffset() const
+{
+    auto orderXValue = hasAttribute(SVGNames::orderAttr) ? orderX() : 3;
+    auto targetXValue = hasAttribute(SVGNames::targetXAttr) ? targetX() : static_cast<int>(floorf(orderXValue / 2));
+    return targetXValue >= 0 && targetXValue < orderXValue;
+}
+
+bool SVGFEConvolveMatrixElement::isValidTargetYOffset() const
+{
+    auto orderYValue = hasAttribute(SVGNames::orderAttr) ? orderY() : 3;
+    auto targetYValue = hasAttribute(SVGNames::targetYAttr) ? targetY() : static_cast<int>(floorf(orderYValue / 2));
+    return targetYValue >= 0 && targetYValue < orderYValue;
+}
+
 void SVGFEConvolveMatrixElement::svgAttributeChanged(const QualifiedName& attrName)
 {
-    switch (attrName.nodeName()) {
```

Comment:
There is no reason to drop the switch statement and use multiple if conditions here. Why make this change?

Reply:
There were a few mistakes there on my part. The main mistake was accidentally having two commits on one PR. So I've fixed that. I will address your concern in the next PR for those specific changes.